fasteners/threads.scad

```diff
@@ -44,32 +59,140 @@ use <scad-utils/transformations.scad>
-// n_starts - Number of thread starts (e.g., DNA, a "double helix," has
-//            n_starts=2).  See wikipedia Screw_thread.
+// Metric Threads in accordance with ISO 68-1
+// ----------------------------------------------------------------------------
+// diameter  - nominal diameter of thread in mm. (e.g. 10 for an M10 thread)
+// pitch     - distance in mm from the crest of one thread to the next.
+// length    - overall axial length of thread in mm.
+// internal  - true = clearances for internal thread (e.g., a nut).
+//             false = clearances for external thread (e.g., a bolt).
+//             (Internal threads should be "cut out" from a solid using
+//             difference()).
+// clearance - additional clearance to add in mm. Teeth are effectively
+//             shifted inwards (outwards) for external (internal) threads by
+//             this amount.
+// n_starts  - Number of thread starts (e.g., DNA, a "double helix," has
+//             n_starts=2).  See Wikipedia Screw_thread.
 module metric_thread (
     diameter = 8,
     pitch = 1,
     length = 1,
     internal = false,
+    clearance = 0,
     n_starts = 1
 )
 {
+    // Tolerancing as per ISO 286
+    // See https://en.wikipedia.org/wiki/IT_Grade
+    function IT_grade (diameter, ITG) =
+        assert(ITG>=1, "ITG has to be between 1 and 16")
+        assert(ITG<=16, "ITG has to be between 1 and 16")
+        let (i = 0.45 * pow(diameter, 1/3) + 0.001 * diameter)
+            pow(10, 0.2*(ITG-1)) * i / 1000;
+
+    // Fundamental deviations as per ISO 286. Formulas obtained from:
+    // https://www.cobanengineering.com/Tolerances/ToleranceBand.asp
+    function EI_es_g (diameter) = 2.5 * pow(diameter, 0.34) / 1000;
+
+    theta = 60;                // V-shape angle, 60º for metric threads
+    H = pitch/2/tan(theta/2);  // height of the V-shape
+    tolerance = internal
+        ? IT_grade(diameter, 6)/2 + 0                  // H6
+        : IT_grade(diameter, 6)/2 + EI_es_g(diameter); // g6
+    D_maj = internal
+        ? diameter + tolerance + clearance
+        : diameter - tolerance - clearance;
+    D_min = D_maj - 2*5/8*H;
+
     trapezoidal_thread (
         pitch = pitch,
         length = length,
         upper_angle = 30, lower_angle = 30,
         outer_flat_length = pitch / 8,
-        major_radius = diameter / 2,
-        minor_radius = diameter / 2 - 5/8 * cos(30) * pitch,
+        major_radius = D_maj / 2,
+        minor_radius = D_min / 2,
         internal = internal,
         n_starts = n_starts
     );
 }
 
+/**
+ * Applies chamfers to a thread object. Expects 2 or 3 children.
+ *
+ * Children:
+ *   1) thread object
+ *   2) chamfer object at near end (Z = 0)
+ *   3) (optional) chamfer object at far end (Z = length)
+ * Parameters:
+ *   length   - axial length in mm of the thread object.
+ *   internal - true when the thread is internal, false when external.
+ */
+module chamfered_thread (length, internal = false)
```

Comment:
Though I named this module `chamfered_thread` now that I revisit it, it doesn't necessarily need the first object to be a thread. So it could be renamed to i.e. `chamfered_cylinder`. Tough maybe too close in naming to the `chamfer_cylinder` module below. Also I'd be happy to move these modules elsewhere out of the `fasteners`/`threads` file/folder if you consider them generic enough to belong elsewhere.

Reply:
I just made a `fillets` folder, this seems like it could go in there.